bricks/distributed/test.cc

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2014 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+#include "../../3rdparty/gtest/gtest-main.h"
+#include "../../bricks/time/chrono.h"
+#include "vector_clock.h"
+
+TEST(VectorClock, SmokeTest) {
+  auto v = VectorClock();
+  v.step();
+
+  // test merge from future - should return false
+  auto data = DiscreteClocks();
+  data.reset(1);
+  EXPECT_EQ(false, v.merge(data));
+  EXPECT_EQ(v.state().size(), (size_t)(1));
+
+  // test lte (t==t')
+  auto v2 = VectorClock(data, data.size());
+  EXPECT_EQ(true, v2.merge(data));
+}
```

Comment:
This test can be cleaner, IMHO.

The constructor with two arguments could be a single-argument constructor, or some `auto v2 = VectorClock::FromVector(data);`

The `data.reset(1)` part above could also be simplified.

Plus, it would help the reader to a) show some `.ToString()`-ified representation (i.e. `EXPECT_EQ()` it), and b) show what minor changes to the input(s) result in the `true` becoming `false` and vice versa.